js/fetch_types.d.ts

```diff
@@ -0,0 +1,426 @@
+type HeadersInit = Headers | string[][] | Record<string, string>;
```

Comment:
Where is this file from? It should probably have a copyright notice?
But mostly I'm asking because I don't want to review it...

Reply:
Bits are copied from typescript's `lib.dom.d.ts` .. I've added a msft copyright header